src/ai_hawk/job_manager.py

```diff
@@ -44,6 +45,7 @@ def _read_env_key_bool(key: str) -> bool:
         return value
 
 
+# TODO: refactor this class for better separation of concern and improved maintainability (God Class anti-pattern)
```

Comment:
create a GitHub issue for this, remove Todo & FixMe.

Applies for all subsequent areas.

Reply:
I think that would take more of a group effort. I left comments like that more as a reminder to create a plan first. It's hard to think about what classes should be created if we don't collectively agree first. 

Should we make some simple design docs first before trying to make significant code changes?

Reply:
yes, it obviously group effort. The first step for this would be filing GitHub enhancement requests (code refactoring), then we can discuss everything over there. there are many TODO items, it won't be one person doing them